### PR TITLE
Add an `sslkeylogfile` parameter to store the TLS master secrets

### PR DESCRIPTION
**Pull request by L3n41c**
### What does this PR do?

Add a `sslkeylogfile` config parameter and its corresponding `DD_SSLKEYLOGFILE` environment variable to specify a file to store the TLS secrets used by the agent.
This file can then be used by wireshark to decipher the encrypted traffic.

### Motivation

There are classes of issues that require to inspect the network traffic for troubleshooting.
Typically, when an agent sends a payload to an intake and the intake rejects that payload, we need to have a way to look at what is precisely the rejected query.

### Additional Notes

This option compromises security and should only be used for debugging.

The name of the environment variable has been chosen to be consistent with other software modulo the `DD_` prefix.
Firefox, Chrome and curl are all 3 using `SSLKEYLOGFILE` environment variable to dump the TLS secrets.

### Describe how to test/QA your changes

Check that the [“Dump the encrypted traffic generated by a datadog agent” howto](https://datadoghq.atlassian.net/l/c/xVLQ9i1C) works as expected.

### Reviewer's Checklist

- [X] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [X] The appropriate `team/..` label has been applied, if known.
- [X] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [X] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [X] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [X] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [X] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/config/config_template.yaml [hunk 1]
```diff
@@ -50,6 +50,17 @@ api_key:
 #
 # skip_ssl_validation: false
 
+## @param sslkeylogfile - string - optional - default: ""
+## @env DD_SSLKEYLOGFILE - string - optional - default: ""
+## sslkeylogfile optionally specifies a destination for TLS master secrets
```
Comment (ruthnaebeck) [inline pkg/config/config_template.yaml:55]:
```suggestion
## sslkeylogfile specifies a destination for TLS master secrets
```

File: pkg/config/config_template.yaml [hunk 2]
```diff
@@ -50,6 +50,17 @@ api_key:
...
+## in NSS key log format that can be used to allow external programs
```
Comment (ruthnaebeck) [inline pkg/config/config_template.yaml:56]:
```suggestion
## in NSS key log format to allow external programs
```

File: pkg/config/config_template.yaml [hunk 3]
```diff
@@ -50,6 +50,17 @@ api_key:
...
+## such as Wireshark to decrypt TLS connections.
+## See https://developer.mozilla.org/en-US/docs/Mozilla/Projects/NSS/Key_Log_Format.
```
Comment (ruthnaebeck) [inline pkg/config/config_template.yaml:58]:
```suggestion
## For more details, see https://developer.mozilla.org/en-US/docs/Mozilla/Projects/NSS/Key_Log_Format.
```

**Review by ruthnaebeck — CHANGES_REQUESTED** [review]
Docs review

**Review by clamoriniere — APPROVED** [review]

**Review by tetrillard-datadog — APPROVED** [review]

**Review by ruthnaebeck — APPROVED** [review]
👍 for docs

**Review by djmitche — APPROVED** [review]
Looks good.  I appreciate the acknowledgement that this compromises security.  Any case where an attacker could modify this setting are cases where the agent is already compromised, so this does not increase the attack surface or introduce a new vulnerability.

**Pull request merged**